kernel/src/hil/aes.rs

```diff
@@ -0,0 +1,14 @@
+use returncode::ReturnCode;
+
+pub trait AESDriver {
+    fn init(&self);
+    fn set_key(&self, key: &'static mut [u8], len: u8);
```

Comment:
Not totally sure I understand, but I think I agree :)

I think that's a fine interface (for now at least) for the syscall driver. Internally, it would be ideal to guarantee at compile-time that the user of the HIL is passing in the right key length.